runelite-client/src/main/java/net/runelite/client/plugins/specialcounter/SpecialCounterPlugin.java

```diff
@@ -474,8 +474,8 @@ private PlayerInfoDrop createSpecInfoDrop(SpecialWeapon weapon, int hit, int pla
 
 	private int getHitDelay(SpecialWeapon specialWeapon, Actor target)
 	{
-		// DORGESHUUN_CROSSBOW is the only ranged wep we support, so everything else is just melee and delay 1
-		if (specialWeapon != SpecialWeapon.DORGESHUUN_CROSSBOW || target == null)
+		// ACCURSED_SCEPTRE and DORGESHUUN_CROSSBOW are the only ranged wep we support, so everything else is just melee and delay 1
```

Comment:
Since it's no longer a single special case, this should be a flag on the weapon and be checked here, eg:

```java
if (!specialWeapon.isRanged() || target == null)
```

Or perhaps the whole method for calculating delay should be a field of the weapon instead of written out in this file.

Reply:
I've went ahead with just adding an isRanged flag to Special Weapon

Reply:
> Since it's no longer a single special case, this should be a flag on the weapon and be checked here, eg:
> 
> ```
> if (!specialWeapon.isRanged() || target == null)
> ```
> 
> Or perhaps the whole method for calculating delay should be a field of the weapon instead of written out in this file.

Just having an `isRanged` seems worse than the original code because it becomes extremely non-obvious that setting isRanged=true requires also adding a case to the switch here. I would either keep the original code or move the cycle computation to a method on the enum.